### PR TITLE
Update read_string method in Process

### PR DESCRIPTION
**Pull request by Gand3lf**
In this way it is also possible to use read_string method based on the string termination character ('\0') without specifying the string length.

### CODE REVIEW

**Comment by MarioVilas** [comment]
Nice! 😊

May I suggest as an optimization to try and read whole pages instead of one
byte at a time, to cut down on the number of syscalls? If you don’t have
the time to implement this, I won’t blame you 😄 I haven’t had the time
myself to implement cool stuff for ages now...
-- 
“There's a reason we separate military and the police: one fights the enemy
of the state, the other serves and protects the people. When the military
becomes both, then the enemies of the state tend to become the people.”

**Comment by Gand3lf** [comment]
Totally agree with you :)

File: winappdbg/process.py [hunk 1]
```diff
@@ -1936,11 +1936,18 @@ def read_string(self, lpBaseAddress, nChars, fUnicode = False):
...
-            while tmp != '\0':
-                szString += tmp
-                tmp = self.read(lpBaseAddress, 1)
-                lpBaseAddress += 1
+            memory = []
+            for mbi in self.get_memory_map(lpBaseAddress, lpBaseAddress):
+                if mbi.State == win32.MEM_COMMIT and \
+                                            not mbi.Protect & win32.PAGE_GUARD:
+                    memory.append( (mbi.BaseAddress, mbi.RegionSize) )
+            lpBasePage, sizePage = memory[0]
```
Comment (MarioVilas) [inline winappdbg/process.py:1944]: Wouldn't this cause an IndexError if lpBaseAddress has a bad address? I'd rather have the right type of exception...

Comment (MarioVilas) [reply winappdbg/process.py:1944]: I'm also thinking if a string is between two pages this would truncate the strings

Comment (Gand3lf) [reply winappdbg/process.py]: You're right. 
What about the new one?

**Comment by MarioVilas** [comment]
Great! 👍 :)

**Pull request merged**